src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Flow.java

```diff
@@ -2111,9 +2111,10 @@ public class AssignAnalyzer extends BaseAnalyzer {
         final Bits uninitsWhenTrue;
         final Bits uninitsWhenFalse;
 
-        /** A mapping from addresses to variable symbols.
+        /** A mapping from addresses to variable symbols, and their diagnostic
+         *  position.
          */
-        protected JCVariableDecl[] vardecls;
+        protected VarAndDeclarationTree[] vars;
```

Comment:
So... normally, a `JCVariableDecl` would have a symbol attached, which is how the old code used to work. Is it correct that this change is needed because the `componentLocalVariables` list is a list of symbols and not of JCVariableDecl? Would it be too complex to create a declaration in Attr and save those in the derived creation tree?

Reply:
Generating the `JCVariableDecl`s in Attr in https://github.com/openjdk/jdk/pull/18509/commits/146513580f96abd5feb7886bd1191805cc93403b.
Thanks for the comment!